job/server/src/test/java/alluxio/job/plan/transform/format/JobPathTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.job.plan.transform.format;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+import alluxio.client.ReadType;
+import alluxio.conf.PropertyKey;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({FileSystem.class, UserGroupInformation.class})
+public class JobPathTest {
+
+  @Before
+  public void before() throws Exception {
+    mockStatic(UserGroupInformation.class);
+
+    when(UserGroupInformation.getCurrentUser()).thenReturn(null);
+  }
+
+  @Test
+  public void testCache() throws Exception {
+    mockStatic(FileSystem.class);
+
+    Configuration conf = new Configuration();
+    JobPath jobPath = new JobPath("foo", "bar", "/baz");
```

Comment:
I'm not sure how this test works. Shouldn't there be some verification that the same JobPath scheme and auth and conf get the same client back, and also, when, scheme, and/or auth, and/or conf is different, we get a different client back? We should also test that if a non-inspected conf is changed, it still returns a cached client.

Reply:
The main reason I did it this way is because I had to mock FileSystem so instead of checking what I get back. I instead check the number of calls to the FileSystem.get

Reply:
We also do test that is a non-inspected conf is changed, it doesn't call the underlying FileSystem.get which is the same thing.

Reply:
Got it. Can you also add a case where there is a new `JobPath` instance, with the same path (fs should be cached), and a new `JobPath` instance with different path (fs should not be cached)